### PR TITLE
cnf-tests: Add numa tests with multiple interfaces

### PR DESCRIPTION
**Pull request by gkopels**
Test cases uses a set SriovNetworkNodePolicies and performance profile to control NUMA node placement in test.
This test creates a pod with sriov interface on NUMA node 1 with the sriov interface in NUMA node 0. The excludeTopology is set to True. THe pod is expected to be deployed.

### CODE REVIEW

File: cnf-tests/testsuites/e2esuite/dpdk/numa_node_sriov.go [hunk 1]
```diff
@@ -62,39 +67,52 @@ var _ = Describe("[sriov] NUMA node alignment", Ordered, func() {
 		sriovDevices, err := sriovCapableNodes.FindSriovDevices(testingNode.Name)
 		Expect(err).ToNot(HaveOccurred())
 
-		numa0Device, err := findDeviceOnNUMANode(testingNode, sriovDevices, "0")
+		numa0DeviceList, err = findDevicesOnNUMANode(testingNode, sriovDevices, "0")
 		Expect(err).ToNot(HaveOccurred())
-		By("Using NUMA0 device " + numa0Device.Name)
+		By("Using NUMA0 device1 " + numa0DeviceList[0].Name)
+		By("Using NUMA0 device2 " + numa0DeviceList[1].Name)
```
Comment (zeeke) [inline cnf-tests/testsuites/e2esuite/dpdk/numa_node_sriov.go:73]: Please, move it in the  `if len(numa0DeviceList) > 1 {` body

File: cnf-tests/testsuites/e2esuite/dpdk/numa_node_sriov.go [hunk 2]
```diff
@@ -142,26 +160,105 @@ var _ = Describe("[sriov] NUMA node alignment", Ordered, func() {
...
+		pod, err := client.Client.Pods(sriovnamespaces.Test).
+			Create(context.Background(), pod, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		Eventually(func(g Gomega) {
+			actualPod, err := client.Client.Pods(sriovnamespaces.Test).Get(context.Background(), pod.Name, metav1.GetOptions{})
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(actualPod.Status.QOSClass).To(Equal(corev1.PodQOSGuaranteed))
+			g.Expect(actualPod.Status.Phase).To(Equal(corev1.PodFailed))
+			g.Expect(actualPod.Status.Message).To(ContainSubstring("Resources cannot be allocated with Topology locality"))
```
Comment (zeeke) [inline cnf-tests/testsuites/e2esuite/dpdk/numa_node_sriov.go:207]: Please, assert on `Status.Reason`

```suggestion
			g.Expect(actualPod.Status.Reason).To(Equal("TopologyAffinityError"))

```

**Review by zeeke — COMMENTED** [review]
Left few minor comments. Overall PR looks good to me

File: cnf-tests/testsuites/e2esuite/dpdk/numa_node_sriov.go [hunk 3]
```diff
@@ -62,39 +67,52 @@ var _ = Describe("[sriov] NUMA node alignment", Ordered, func() {
...
 
-		createSriovNetworkAndPolicyForNumaAffinityTest(8, numa1Device, "#4-7",
-			"test-numa-1-exclude-topology-true-", testingNode.Name,
-			"testNuma1ExcludeTopoplogyTrue", ipam, true)
+		createSriovNetworkAndPolicyForNumaAffinityTest(8, numa1DeviceList[0], "#0-3",
+			"test-numa-1-nic1-exclude-topology-false-", testingNode.Name,
+			"testNuma1NIC1ExcludeTopoplogyFalse", ipam, false)
+
+		createSriovNetworkAndPolicyForNumaAffinityTest(8, numa1DeviceList[0], "#4-7",
+			"test-numa-1-nic2-exclude-topology-true-", testingNode.Name,
```
[the remark this answers is not in the document]
Comment (gkopels) [reply cnf-tests/testsuites/e2esuite/dpdk/numa_node_sriov.go:114]: changed

File: cnf-tests/testsuites/e2esuite/dpdk/numa_node_sriov.go [hunk 4]
```diff
@@ -62,39 +67,52 @@ var _ = Describe("[sriov] NUMA node alignment", Ordered, func() {
 		sriovDevices, err := sriovCapableNodes.FindSriovDevices(testingNode.Name)
 		Expect(err).ToNot(HaveOccurred())
 
-		numa0Device, err := findDeviceOnNUMANode(testingNode, sriovDevices, "0")
+		numa0DeviceList, err = findDevicesOnNUMANode(testingNode, sriovDevices, "0")
 		Expect(err).ToNot(HaveOccurred())
```
[the remark this answers is not in the document]
Comment (gkopels) [reply cnf-tests/testsuites/e2esuite/dpdk/numa_node_sriov.go:71]: changed

File: cnf-tests/testsuites/e2esuite/dpdk/numa_node_sriov.go (hunk 2 above)
Comment (gkopels) [reply cnf-tests/testsuites/e2esuite/dpdk/numa_node_sriov.go:207]: changed

File: cnf-tests/testsuites/e2esuite/dpdk/numa_node_sriov.go [hunk 5]
```diff
@@ -142,26 +160,105 @@ var _ = Describe("[sriov] NUMA node alignment", Ordered, func() {
...
-func findDeviceOnNUMANode(node *corev1.Node, devices []*sriovv1.InterfaceExt, numaNode string) (*sriovv1.InterfaceExt, error) {
-	for _, device := range devices {
-		out, err := nodes.ExecCommandOnNode([]string{
-			"cat",
-			filepath.Clean(filepath.Join("/sys/class/net/", device.Name, "/device/numa_node")),
-		}, node)
-		if err != nil {
-			klog.Warningf("can't get device [%s] NUMA node: out(%s) err(%s)", device.Name, string(out), err.Error())
-			continue
+	It("Validate the creation of a pod with excludeTopology set to True and an SRIOV interface in a different NUMA node "+
```
Comment (SchSeba) [inline cnf-tests/testsuites/e2esuite/dpdk/numa_node_sriov.go:164]: do we need to check the cpus that got allocated into the running pod?

**Comment by zeeke** [comment]
/approve
LGTM

@SchSeba there are some pushes after your comment. Please, have a look

File: cnf-tests/testsuites/e2esuite/dpdk/numa_node_sriov.go [hunk 6]
```diff
@@ -139,29 +158,120 @@ var _ = Describe("[sriov] NUMA node alignment", Ordered, func() {
 			g.Expect(actualPod.Status.QOSClass).To(Equal(corev1.PodQOSGuaranteed))
 		}, 30*time.Second, 1*time.Second).Should(Succeed())
 
+		By("Validate Pod NUMA Node")
+		expectPodCPUsAreOnNUMANode(pod, 1)
```
Comment (SchSeba) [inline cnf-tests/testsuites/e2esuite/dpdk/numa_node_sriov.go:162]: I think you need to add this one to the others tests no?
@zeeke WDYT?

Comment (gkopels) [reply cnf-tests/testsuites/e2esuite/dpdk/numa_node_sriov.go:162]: It was added to all testts that didn't fail to deploy a pod.

**Review by SchSeba — APPROVED** [review]
/lgtm
/approve

**Pull request merged**